pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -536,7 +658,11 @@ func (vs *VSphere) DiskIsAttached(volPath string, nodeName k8stypes.NodeName) (b
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 		// Ensure client is logged in and session is valid
-		err := vs.conn.Connect(ctx)
+		vsi, err := vs.getVSphereInstance(nodeName)
+		if err != nil {
+			return false, err
+		}
+		err = vsi.conn.Connect(ctx)
```

Comment:
Just for verification, 
Over here vm, err := vs.getVMByName(ctx, nodeName). If this errors out, you will scan through all the Virtual Centers and will update the cache. Right?

Same question at all other places?

Reply:
Yes, nodemanager.getNodeInfo will have to retry if the node is not discovered yet possibly because the node controller has not notified about that node yet. 

It could so happen that the node manager has the node info in cache but it is stale. In such cases, the caller need to invoke nodemanager.registerNode so that node manager can rediscover the node VM.